master/buildbot/reporters/gerrit_verify_status.py

```diff
@@ -0,0 +1,198 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from twisted.internet import defer
+from twisted.python import failure
+
+from buildbot.process.properties import Interpolate
+from buildbot.process.properties import Properties
+from buildbot.process.results import CANCELLED
+from buildbot.process.results import EXCEPTION
+from buildbot.process.results import FAILURE
+from buildbot.process.results import RETRY
+from buildbot.process.results import SKIPPED
+from buildbot.process.results import SUCCESS
+from buildbot.process.results import WARNINGS
+from buildbot.reporters import http
+from buildbot.util import httpclientservice
+from buildbot.util.logger import Logger
+
+log = Logger()
+
+
+class GerritVerifyStatusPush(http.HttpStatusPushBase):
+    name = "GerritVerifyStatusPush"
+    neededDetails = dict(wantProperties=True)
+    # overridable constants
+    RESULTS_TABLE = {
+        SUCCESS: 1,
+        WARNINGS: 1,
+        FAILURE: -1,
+        SKIPPED: 0,
+        EXCEPTION: 0,
+        RETRY: 0,
+        CANCELLED: 0
+    }
+    DEFAULT_RESULT = -1
+
+    @defer.inlineCallbacks
+    def reconfigService(self, baseURL, auth,
+                        startDescription=None, endDescription=None,
+                        verification_name=None, abstain=False, category=None, reporter=None,
+                        verbose=False, **kwargs):
+        yield http.HttpStatusPushBase.reconfigService(self, **kwargs)
+
+        if baseURL.endswith('/'):
+            baseURL = baseURL[:-1]
+
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, baseURL, auth=auth)
+
+        self.verification_name = verification_name or Interpolate('%(prop:buildername)s')
+        self.reporter = reporter or "buildbot"
+        self.abstain = abstain
+        self.category = category
+        self.startDescription = startDescription or 'Build started.'
+        self.endDescription = endDescription or 'Build done.'
+        self.verbose = verbose
```

Comment:
do we really need this?  with the fine grained control over the logging this seems to be unnecessary [any more]?